dipy/denoise/noise_estimate_localpca.py

```diff
@@ -0,0 +1,111 @@
+"""
+===========================================
+Estimate Noise Levels for LocalPCA Datasets
+===========================================
+
+"""
+
+import numpy as np
+import nibabel as nib
+import scipy as sp
+from scipy import ndimage
+
+def estimate_sigma_localpca(data,gtab):
+    
+    # first identify the number of the b0 images
+    K = gtab.b0s_mask[gtab.b0s_mask].size
+
+    if(K>1):
+        print("MUBE Estimate")
+        # If multiple b0 values then 
+        data0 = data[...,gtab.b0s_mask]
+        # MUBE Noise Estimate
+
+        # Form their matrix
+        X = data0.reshape(data0.shape[0] * data0.shape[1] * data0.shape[2], data0.shape[3])
+        # X = NxK
+        # Now to subtract the mean
+        M = np.mean(X,axis = 0)
+        X = X - M
+        C = np.transpose(X).dot(X)
+        # C = C/data0.shape[3]
+        # Do PCA and find the lowest principal component
+        [d,W] = np.linalg.eigh(C)
+        d[d < 0] = 0;
+        d_new = d[d != 0]
```

Comment:
I know that we discuss this in the meeting, however I do not get it - if the number of voxels per volume is larger than the number of volumes (i.e. X has a dimention of Nvoxels x Nvolumes), you will not have zero eigenvalues (assuming that all volumes contain different information). So you basically have to use the lower eigenvalue to have the PCA lower component as mentioned in the article (a suggestion - plot the PCA components to check if this only contains noise as I am assuming).